packages/sdk-react/src/components/receivables/InvoicesTable/useInvoiceRowActionMenuCell.tsx

```diff
@@ -0,0 +1,308 @@
+    id: string;
+    action?: InvoicesTableRowAction;
+  }) => void;
+
+  /**
+   * The actions to be displayed in the row's action menu.
+   * If not provided, default actions will be displayed,
+   * in case if `onRowActionClick` is provided.
+   *
+   * @example
+   * ```ts
+   * {
+   *   "draft": ['view', 'edit', 'issue', 'delete'], // Optional
+   *   "issued": ['view', 'send', 'cancel', 'copyPaymentLink'], // Optional
+   *   "canceled": ['view', 'send'], // Optional
+   *   "partially_paid": ['view', 'send', 'copyPaymentLink', 'cancel'], // Optional
+   *   "overdue": ['view', 'send', 'copyPaymentLink', 'cancel', 'markUncollectible'], // Optional
+   *   "paid": ['view'], // Optional
+   *   "uncollectible": ['view'], // Optional
+   *   "expired": ['view'], // Optional
+   *   "accepted": ['view'], // Optional
+   *   "declined": ['view'], // Optional
+   *   "recurring": ['view'], // Optional
+   *   "deleted": [], // Optional
+   * }
+   * ```
+   */
+  rowActions?: Partial<InvoicesTableRowActionSchema>;
+}
+
+export const useInvoiceRowActionMenuCell = (
+  props: UseInvoiceRowActionMenuCellProps | {}
+): GridActionsColDef<ReceivableResponse> | undefined => {
+  const { data: receivableActionSchema, userIdFromAuthToken } =
+    usePermissions('receivable');
+  const { i18n } = useLingui();
+
+  if (!('onRowActionClick' in props && props.onRowActionClick)) return;
+
+  return {
+    field: 'action_menu',
+    type: 'actions',
+    headerName: t(i18n)({
+      message: 'Action menu',
+      context: 'InvoicesTableRowActionMenu',
+    }),
+    renderHeader: () => null,
+    getActions: (params) => {
+      const menuItems = getInvoiceActionMenuItems({
+        // casts, because it is not possible to resolve `type` based on multiple different enum
+        invoice: params.row as InvoiceResponsePayload,
+        actions: props.rowActions,
+        receivableActionSchema,
+        userIdFromAuthToken,
+        i18n,
+      });
+
+      return menuItems.map(({ label, action }) => (
+        <GridActionsCellItem
+          showInMenu
+          label={label}
+          onClick={(event) => {
+            event.preventDefault();
+            props.onRowActionClick({ id: params.row.id, action });
+          }}
+        />
+      ));
+    },
+  };
+};
+
+const getInvoiceActionMenuItems = ({
+  invoice,
+  actions: inActions,
+  receivableActionSchema,
+  userIdFromAuthToken,
+  i18n,
+}: {
+  invoice: InvoiceResponsePayload;
+  actions?: Partial<InvoicesTableRowActionSchema>;
+  receivableActionSchema: ActionSchema[] | undefined;
+  userIdFromAuthToken: string | undefined;
+  i18n: I18n;
+}): InvoicesTableRowActionMenuItem[] => {
+  const actions =
+    inActions?.[invoice.status] ?? DEFAULT_ACTION_LIST[invoice.status];
+
+  const allowedActions = filterInvoiceActionMenuAllowedItems(
+    receivableActionSchema,
+    actions,
+    invoice,
+    userIdFromAuthToken
+  );
+
+  const allMenuItems = getInvoiceActionMenuItemLabels(i18n);
```

Comment:
Maybe we should memoize the result from this function? As I can see we call it on each click but we don't need to create a new object every time when the user clicks on an item. We could safe the returned value (especially because `i18n` never changes) and reuse the result without generating a new object

Reply:
I don't think that it is so needed:

```
1. getInvoiceActionMenuItems: 0.0068359375 ms
2. getInvoiceActionMenuItems: 0.006103515625 ms
3. getInvoiceActionMenuItems: 0.005859375 ms
4. getInvoiceActionMenuItems: 0.005126953125 ms
5. getInvoiceActionMenuItems: 0.006103515625 ms
6. getInvoiceActionMenuItems: 0.005859375 ms
7. getInvoiceActionMenuItems: 0.006103515625 ms
8. getInvoiceActionMenuItems: 0.0048828125 ms
9. getInvoiceActionMenuItems: 0.006103515625 ms
10. getInvoiceActionMenuItems: 0.005126953125 ms
```

JS is just fast enough with the operations on simple data structures.